### ISSUE TITLE
CassQuery.getSingleResult returns empty list when item not found

### ISSUE THREAD

**Issue by bramp** [Impetus/kundera#509]
From what I understand about the JPA, getSingleResult is meant to throw an NoSuchElementException if there is no single result. Instead it returns an empty list. Additionally it should throw NonUniqueResultException if more than one element is found. It throws neither.

```
    @Override
    public Object getSingleResult()
    {
        // to fetch a single result form database.
        isSingleResult = true;
        List results = getResultList();
        isSingleResult = false;
        return results.isEmpty() ? results : results.get(0);
    }
```

**Comment by mevivs** [comment]
Fix  added on current trunk and releasing with 2.10

-Vivek

**Comment by bramp** [comment]
Awesome thanks.

**Comment by mevivs** [comment]
Fixed and releasing with 2.10

-Vivek
